src/lib/phdi-building-blocks/phdi_building_blocks/linkage.py

```diff
@@ -1,68 +1,100 @@
-            address_line = ""
-            if "address" in patient:
-                address = next(
-                    (addr for addr in patient["address"] if addr.get("use") == "home"),
-                    patient["address"][0],
-                )
-                address_line = " ".join(address.get("line", []))
-                address_line += f" {address.get('city')}, {address.get('state')}"
-                if "postalCode" in address and address["postalCode"]:
-                    address_line += f" {address['postalCode']}"
+        # Generate and store unique hash code
+        link_str = name_str + "-" + patient["birthDate"] + "-" + address_line
+        hashcode = generate_hash_str(link_str, salt_str)
 
-            # Generate and store unique hash code
-            link_str = name_str + "-" + patient["birthDate"] + "-" + address_line
-            hashcode = generate_hash_str(link_str, salt_str)
+        if "identifier" not in patient:
+            patient["identifier"] = []
 
-            if "identifier" not in patient:
-                patient["identifier"] = []
+        patient["identifier"].append(
+            {
+                "value": hashcode,
+                # Note: this system value corresponds to the FHIR specification
+                # for a globally used / generated ID or UUID--the standard here
+                # is to make the use "temporary" even if it's not
+                "system": "urn:ietf:rfc:3986",
+                "use": "temp",
+            }
+        )
+    return bundle
 
-            patient["identifier"].append(
-                {
-                    "value": hashcode,
-                    # Note: this system value corresponds to the FHIR specification
-                    # for a globally used / generated ID or UUID--the standard here
-                    # is to make the use "temporary" even if it's not
-                    "system": "urn:ietf:rfc:3986",
-                    "use": "temp",
-                }
-            )
+
+def get_field_with_use(patient: dict, field: str, use: str, default_field: int) -> str:
+    """
+    For a given field (such as name or address), find the first-occuring
+    instance of the field in a given patient JSON dict such that the
+    instance is associated with a particular use. I.e. find the first
+    name for a patient that has an "official" use capacity. If no
+    instance of a field with the requested use case can be found, instead
+    return a specified default field.
+    :param dict patient: Patient from a FHIR bundle
+    :param str field: The field to extract
+    :param str use: The use the field must have to qualify
+    :param int default_field: The index of the field type to treat as
+    the default return type if no field with the requested use case is
+    found
+    :return: The requested use-case-type field
+    :rtype: str
+    """
+    return next(
+        (item for item in patient[field] if item.get("use") == use),
```

Comment:
This assumes that patient[field] exists as well as patient[field][default_field] else I think it will throw a KeyError.  I don't think this is ideal?  If we decide to use this code as-is, we should document this behavior better.

Reply:
This will be added as a TODO, which will get turned into a ticket quickly after this PR lands.